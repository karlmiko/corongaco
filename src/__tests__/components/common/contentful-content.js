import React from 'react'
import renderer from 'react-test-renderer'
import { useStaticQuery } from 'gatsby'
import ContentfulContent from '~components/common/contentful-content'

beforeEach(() => {
  useStaticQuery.mockImplementation(() => ({
    site: {
      siteMetadata: {
        contentfulSpace: '1234',
      },
    },
  }))
})

describe('Components : Common: Contentful content', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ContentfulContent content="<p>This is a paragraph</p>" id="1234" />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with an editor', () => {
    Object.defineProperty(window, 'location', {
      value: {
        search: '?edit',
      },
      writable: true,
    })
    const tree = renderer
      .create(
        <ContentfulContent content="<p>This is a paragraph</p>" id="1234" />,
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
    expect(tree.find(a => a.type === 'a')).toBeTruthy()
  })

  it('renders correctly without an editor', () => {
    Object.defineProperty(window, 'location', {
      value: {
        search: '?q',
      },
      writable: true,
    })
    const tree = renderer
      .create(
        <ContentfulContent content="<p>This is a paragraph</p>" id="1234" />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
