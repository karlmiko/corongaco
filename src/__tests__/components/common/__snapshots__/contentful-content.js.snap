// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Common: Contentful content renders correctly 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>This is a paragraph</p>",
    }
  }
/>
`;

exports[`Components : Common: Contentful content renders correctly with an editor 1`] = `
Array [
  <a
    className="edit"
    href="https://app.contentful.com/spaces/1234/entries/1234"
    rel="noopener noreferrer"
    target="_blank"
  >
    Edit this content
  </a>,
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>This is a paragraph</p>",
      }
    }
  />,
]
`;

exports[`Components : Common: Contentful content renders correctly without an editor 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>This is a paragraph</p>",
    }
  }
/>
`;
